### DISCUSSION TITLE
Entity with field storing filter expressions

### DISCUSSION THREAD

**Discussion by thaingo** [yahoo/elide#2399] [Q&A]
I have a use-case in which api client wants to store a condition in an entity field. The condition is a combination of different `filter expressions`. At one point in time, the condition will be validated by backend side. If the validation result is `true` then backend will take some action, otherwise other one. 

To leverage Elide and its rsql parser, here is what in my mind for now:

1. The condition is stored as a string and it would be [filterParams](https://github.com/yahoo/elide/blob/27d2e82bbcd79d197265b6e7dd976344f02e43ed/elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java#L139)
2. And the condition supports only [filter expression for specific collection types](https://github.com/yahoo/elide/blob/27d2e82bbcd79d197265b6e7dd976344f02e43ed/elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java#L75)
3. The condition will be validated at runtime or at some scheduled time by backend.

To make the idea happens, I need to have ability to parse the condition (i), convert the condition to, let's say JPQL/HQL (ii), execute the queries and get result (iii), and then check the result to take some action (iv). Could you please advise me, especially item (i) and (ii)?

**Comment by aklish** [comment]
(i) is done in Elide in the `FilterTranslator`:
https://github.com/yahoo/elide/blob/master/elide-datastore/elide-datastore-jpql/src/main/java/com/yahoo/elide/datastores/jpql/filter/FilterTranslator.java#L296-L298

This code is used both by JPA and aggregation stores to generate JPQL or SQL from a filter predicate.

(ii) is done in the JPA/JPQL store here:
https://github.com/yahoo/elide/blob/master/elide-datastore/elide-datastore-jpql/src/main/java/com/yahoo/elide/datastores/jpql/query/RootCollectionFetchQueryBuilder.java
